api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -238,6 +238,62 @@ message HttpConnectionManager {
     google.protobuf.BoolValue enabled = 3;
   }
 
+  // [#not-implemented-hide] Transformations that apply to path
+  // headers. Transformations are applied before any processing of requests by
+  // HTTP filters, routing, and matching. Only the normalized path will be
+  // visible internally if a transformation is enabled. The policy determines
+  // whether Envoy will forward the normalized *:path* header or forward the
+  // original header *:path* header (unless a filter modifies it). Note:
+  // observing (logging, tracing) headers after a transformation is applied will
+  // show the normalized header. Observing after headers are sent to the codec
+  // for encoding to the upstream (e.g. when closing the stream after a response
+  // is received from an upstream) will show the forwarded header.
+  message PathNormalizationOptions {
+    // Defines a policy to apply to a normalization. Any path rewrites that the
+    // router performs (e.g. :ref:`regex_rewrite
+    // <envoy_api_field_config.route.v3.RouteAction.regex_rewrite>` or
+    // :ref:`prefix_rewrite
+    // <envoy_api_field_config.route.v3.RouteAction.prefix_rewrite>`) will apply
+    // to the *:path* header destined for the upstream.
+    enum NormalizationPolicy {
```

Comment:
Do we ever envision having params for any of this? I'm worried that enum here might be fragile. Should we instead do a oneof with a message for each option, even if that message is empty for now? This would be easier to future proof IMO?

Reply:
Yeah, @adisuissa brought that up. I'll change to message per option (empty if no params). 

By the way, @yangminzhu has been creating a proposal to add these types of transformations (ModSecurity based https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual-(v2.x)#Transformation_functions) to `StringMatcher` as well so that matching can be done in filters with added transformations (e.g. we already have the `lowercase` transformation with `ignore_case` in StringMatcher).

I was planning to change this after talking with @yangminzhu to define a core message `TextTransformations` with repeated operations (a oneof with message per option), and consuming that here for a list of internal transformations and forwarding transformations?